src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -148,6 +216,56 @@ private static bool Quit(string title, string text)
             return n == 0;
         }
 
+        private void FilterData(ApplicationData applicationData, string filter, int[] columnWidths, int offset, Label filterError)
+        {
+            var items = new List<string>();
+            filterError.Text = " ";
+            filterError.ColorScheme = Colors.Base;
+            filterError.Redraw(filterError.Bounds);
+
+            foreach (DataTableRow dataTableRow in applicationData.DataTable.Data)
+            {
+                bool match = false;
+                var valueList = new List<string>();
+                foreach (var dataTableColumn in applicationData.DataTable.DataColumns)
+                {
+                    string dataValue = dataTableRow.Values[dataTableColumn.ToString()].DisplayValue;
+
+                    if (!match)
+                    {
+                        try
+                        {
+                            if (Regex.IsMatch(dataValue, filter, RegexOptions.IgnoreCase))
+                            {
+                                match = true;
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            filterError.Text = ex.Message;
+                            filterError.ColorScheme = Colors.Error;
+                            filterError.Redraw(filterError.Bounds);
+                            return;
+                        }
+                    }
+                    
+                    valueList.Add(dataValue);
+                }
+
+                if (match)
+                {
+                    items.Add(GetPaddedString(valueList, columnWidths, offset));
+                }
+            }
+
+            _itemList = items;
+        }
+
+        private void FilterField_Changed(object sender, ustring e)
+        {
+            // this doesn't get hit when field text is changed?
```

Comment:
This will just run `FilterData` yeah? Maybe worth putting the code in and commenting it out.

and also put the issue number as we're not sure when that commit will be in a release.

Reply:
Yes, but once this event works, we should remove the Apply button as it's implicit.  I can add the code and a comment.